applications/app/controllers/InteractiveController.scala

```diff
@@ -5,21 +5,49 @@ import common._
 import contentapi.ContentApiClient
 import conf._
 import conf.switches.Switches
+import model.Cached.RevalidatableResult
 import model._
+import play.api.libs.iteratee.Enumerator
+import play.api.libs.ws.{WSResponse, WSClient}
 import play.api.mvc._
 import views.support.RenderOtherStatus
+import conf.Configuration.interactive.cdnPath
+import conf.Configuration.environment.isPreview
+import scala.concurrent.duration._
 
 import scala.concurrent.Future
 
 case class InteractivePage (interactive: Interactive, related: RelatedContent) extends ContentPage {
   override lazy val item = interactive
 }
 
-class InteractiveController(contentApiClient: ContentApiClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
+class InteractiveController(contentApiClient: ContentApiClient, wsClient: WSClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
 
   def renderInteractiveJson(path: String): Action[AnyContent] = renderInteractive(path)
   def renderInteractive(path: String): Action[AnyContent] = Action.async { implicit request => renderItem(path) }
 
+  def proxyInteractiveServiceworker(path: String): Action[AnyContent] = Action.async { implicit request =>
+    val stage = if (isPreview) "preview" else "live"
+    val serviceWorkerPath = s"$cdnPath/service-workers/$stage/$path/interactive-service-worker.js"
+    /*
+      This just passes through the http response from the above url so we don't need to
+      match status codes etc and setting the correct cache headers is handled by Cached()
+     */
+
+    wsClient.url(serviceWorkerPath).get().map { response =>
+      Cached (7 days) {
+        RevalidatableResult(convertWSResponseToResult(response), response.body)
+      }
+    }
+  }
+
+  def convertWSResponseToResult(response: WSResponse): Result = {
+    val headers = response.allHeaders.map { case (header, body) =>
```

Comment:
extremely nervous about allowing arbitrary headers through, we very nearly let through an HSTS header when we proxied discussion through theguardian.com as a temporary measure and that would have been super hard to get out of.  Obv not an issue now, but still.